File: 04 - ORIENTAÇÃO A OBJETOS/nodeJS - JavaScript_/Exercicio 1/Fazendo com classes/Exercicio 1.js
class CaixaDeNumero {
    constructor(numeroSecreto){
        this.numeroSecreto = numeroSecreto
    }

    ehPar(){
        return this.numeroSecreto % 2 == 0
    }

    dizerQualNumero(){
        return this.numeroSecreto
    }

    dizerTabuada(){
        let vetor = []
        for(let i=0; i<10; i++){
            vetor[i] = this.numeroSecreto * (i+1) 
        }
        return vetor
    }
}

//PRINCIPAL
const geraValor = Math.floor(Math.random() * 10)
const caixa = new CaixaDeNumero(geraValor)

if(caixa.ehPar())
    console.log('É par')
else
    console.log('Não é par')

console.log('O número secreto é: '+caixa.dizerQualNumero())

console.log('A tabuada: '+caixa.dizerTabuada())

File: 04-ORIENTACAO_A_OBJETOS/Python/Exercicio 1/Exercicio 1.py
class CaixaDeNumero:
    def __init__(self, numeroSecreto):
        self.numeroSecreto = numeroSecreto

    def ehPar(self):
        return self.numeroSecreto % 2 == 0

    def dizerTabuada(self):
        array = []
        for i in range(10):
            array.append(self.numeroSecreto * (i+1))
        return array

    def dizerNumero(self):
        return self.numeroSecreto

import random
numero = random.randrange(1, 10)
caixa = CaixaDeNumero(numero)

if caixa.ehPar():
    print('O numero é par')
else:
    print('O número é ímpar')

print(caixa.dizerTabuada())

print(caixa.dizerNumero())

    